src/components/rule-builder/rule-builder.component.tsx

```diff
@@ -0,0 +1,1028 @@
+      [conditions, launchDeleteConditionsOrActions],
+    );
+
+    return (
+      <div className={styles.container}>
+        <div className={styles.ruleHeaderContainer}>
+          {!isToggleVisible && (
+            <RuleHeader
+              isRequired={isRequired}
+              isAllowFutureDate={isAllowFutureDate}
+              isDisallowDecimals={isDisallowDecimals}
+              handleRequiredChange={handleRequiredChange}
+              handleAllowFutureDateChange={handleAllowFutureDateChange}
+              handleDisallowDecimalValueChange={handleDisallowDecimalValueChange}
+              ruleId={ruleId}
+              question={question}
+            />
+          )}
+          {!isNewRule && <ConditionalLogicHelpLink />}
+        </div>
+        <div className={styles.ruleBuilderContainer}>
+          <div className={styles.conditionsContainer}>
+            {conditions.map((condition, index) => (
+              <RuleCondition
+                key={condition.id}
+                fieldId={condition.id}
+                questions={questions}
+                answers={answers}
+                isTablet={isTablet}
+                isNewCondition={condition.isNew}
+                isNewRule={isNewRule}
+                deleteCondition={() => deleteCondition(condition.id)}
+                launchDeleteConditionalLogicModal={(deleteAll: boolean) => launchDeleteConditionalLogicModal(deleteAll)}
+                addCondition={addCondition}
+                addNewConditionalLogic={addNewConditionalLogic}
+                handleConditionChange={handleConditionChange}
+                index={index}
+                conditions={conditions}
+              />
+            ))}
+          </div>
+          <div className={styles.actionsContainer}>
+            {actions.map((action, index) => (
+              <RuleAction
+                key={action.id}
+                fieldId={action.id}
+                questions={questions}
+                isTablet={isTablet}
+                isNewAction={action.isNew}
+                deleteAction={() => deleteAction(action.id)}
+                addAction={addAction}
+                handleActionChange={handleActionChange}
+                index={index}
+                actions={actions}
+              />
+            ))}
+          </div>
+        </div>
+      </div>
+    );
+  },
+);
+
+export default RuleBuilder;
+
+export const ConditionalLogicHelpLink = () => {
+  return (
+    <ConfigurableLink to={helpLink} className={styles.helpLink}>
+      <span>
+        <Help />
+      </span>
+      <p>Learn about conditional logic</p>
+    </ConfigurableLink>
+  );
+};
+interface RuleHeaderProps {
+  isRequired: boolean;
+  isAllowFutureDate: boolean;
+  isDisallowDecimals: boolean;
+  handleRequiredChange: () => void;
+  handleAllowFutureDateChange: () => void;
+  handleDisallowDecimalValueChange: () => void;
+  ruleId: string;
+  question: Question;
+}
+export const RuleHeader = React.memo(
```

Comment:
I don't think we should be memoizing all of these components. This one doesn't seem like it has a huge deal of performance overhead. Memoizing prematurely actually contributes to performance bottlenecks. Have you profiled these and found them to be non-performant?